### ISSUE TITLE
Problems when calculating the new boundingbox

### ISSUE THREAD

**Issue by ValorNaram** [babykarte/babykarte.github.io#120]
@Discostu36 catched some serious bug which I want to describe here for you.

## Preparation
Some areas aren't included when you move the map for a certain time to the top, left and top left corners. The bug occurs in `js/overpass.js` in function `locateNewArea´ which is called once the user moves the map for each filter. I guess the bug to be somewhere between line 193 and 199. In OSM terminology the boundingbox consists of the following syntax: `(south, east, north, west)`

```javascript
} else if (east_new - east_old >= accuracy) {
	west_new = east_old;
	if (east_new > maxEast) {
		loadingAllowed = true;
		west_new = maxEast;
		maxEast = east_new;
	}
}
```
## The bug looks like a... oh well... a 🐞
![1](https://user-images.githubusercontent.com/19347636/63190220-126f6c00-c066-11e9-86d5-7de1a35156d4.jpg)
**Pic**<br/>
A screenshot showing the Overpass SQL request to Overpass API. I market there the coordination for the east which Babykarte has calculated and sent to Overpass.

![2](https://user-images.githubusercontent.com/19347636/63190329-4ba7dc00-c066-11e9-9866-e94282825549.jpg)
**Pic 2**<br/>
**Legend**
red area: The area Babykarte should include in its request. That's how users assume that Babykarte does include that area.
blue area: The area Babykarte actually requests data for. That's how Babykarte behaves now.
yellow line: The line which represent the east border.
yellow area in the address bar: Represents the coordinate you've seen in _Pic 1_

**Explanation**
Babykarte calculates the east wrong and that's why it requests data only for the blue area instead of requesting it also for the red area. I guess this problem always persists when moving to the east and not just at a certain amount of time passed away.

**Comment by ValorNaram** [comment]
## Update in bug finding

Beside the fact that I understood and therefore described the bug wrong to you I found out that the bug does exist in every condition from line 151 to 251 (function `locateNewArea` in file `js/overpass.js`)

**Comment by ValorNaram** [comment]
## Update 2 in bug finding


I finally found the bug. It's the whole algorithm. Babykarte thinks to have areas downloaded which actually aren't. Below you will find the visualisation on my whiteboard.
![IMG_20190818_161436](https://user-images.githubusercontent.com/19347636/63225718-9dba4000-c1d3-11e9-9d8f-2791c6aff3c2.jpg)

**Comment by Discostu36** [comment]
Do you know how to fix it?

**Comment by ValorNaram** [comment]
I'm thinking about harmonizing the areas so each area takes the same size. So if the user moves the map, Babykarte will not load only the area in the users view but everything that belongs to this zone. E.g. the user moves the map by 0,050 to the north, Babykarte will load new data for 0,10 to the north (more than the user requested) in order to harmonize and create a grid system.

Like the system for Map Tiles it allows us fixed areas and therefore fixed boundingbox and one cheme we can repeatly use and calculate.

**Comment by ValorNaram** [comment]
The visualisation of what I want to do

![IMG_20190818_200232](https://user-images.githubusercontent.com/19347636/63228527-80499e00-c1f4-11e9-83f2-ff536b729788.jpg)

I will figure out how to start. I need some preparation and thinking where it could be best to start to change code.

**Comment by ValorNaram** [comment]
## Update on bug fixing process

![IMG_20190822_160606](https://user-images.githubusercontent.com/19347636/63521545-f1d86380-c4f6-11e9-8755-4b1baf90f7df.jpg)

I decided to apply the system of Map Tiles. That means not only Map Tiles are divided into squares all of the same sizes but also Map Data (marker and dots). Currently I'm writing the code and after that I will invite you all Babykarte-lovers to test it :1st_place_medal:

**Comment by ValorNaram** [comment]
I uploaded the new algorithm. I am inviting you all to test it :)

Please use this issue to send bugs, questions etc. regarding this new algorithm.

**Comment by Discostu36** [comment]
I cannot really do I detailed debugging now, but it certainly does not work correctly yet. 

1. Queries seem to be to complicated now. Everything takes really long. I now get frequent timeouts, for example when trying to get playgrounds with playground features at lower zoom levels  (and Babykarte is not giving an error message).
2. The bounding box is not calculated correctly. POIs at the edges of the bounding box are often not calculated (example see below)
3. When my IP is blocked for too many requests, Babykarte tells me to reload the map. This will not help.

![grafik](https://user-images.githubusercontent.com/30999326/63722097-c89b3880-c852-11e9-96ec-d8f3203d2826.png)

**Comment by ValorNaram** [comment]
> Queries seem to be to complicated now. Everything takes really long.

Yes, they are. Queries are more complicated and the areas Babykarte requests are large. The size of each squares is `0.02` and that's almost the size of the displayed map.

> The bounding box is not calculated correctly. POIs at the edges of the bounding box are often not calculated (example see below)

Are you sure you have the latest version of `devel`? Because the commit https://github.com/babykarte/babykarte.github.io/commit/ea709a6e73d0f3d417b318a908b093821252fb43 could have fixed it. I had this problem you described and I fixed it by writing my own math rounding function which ignores the mathematical rule `If equal or greater than 5, then the number will be rounded`.

> When my IP is blocked for too many requests, Babykarte tells me to reload the map. This will not help.

Yes, because the requests are larger and more complicated.
